external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaBatch.scala

```diff
@@ -22,12 +22,13 @@ import org.apache.kafka.common.TopicPartition
 import org.apache.spark.SparkEnv
 import org.apache.spark.internal.Logging
 import org.apache.spark.internal.config.Network.NETWORK_TIMEOUT
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
 import org.apache.spark.sql.sources.v2.reader.{Batch, InputPartition, PartitionReaderFactory}
 
 
 private[kafka010] class KafkaBatch(
     strategy: ConsumerStrategy,
-    sourceOptions: Map[String, String],
+    sourceOptions: CaseInsensitiveMap[String],
```

Comment:
shall we use `CaseInsensitiveStringMap` which is used a lot in Data Source V2?

Reply:
Just back from vacation and syncing. So originally I've considered to use `CaseInsensitiveStringMap` but not chosen because of the following:
* `CaseInsensitiveStringMap` implements java `Map` and if we're passing that around then lot of useful feature would disappear (`getOrElse`, `find`, ...) or we just have to add `asScala` to all the places where scala feature are used. All in all converting forth and back was overkill from my point of view.
* `KafkaBatch` is private API and though it can be different due to the considerations above.

As always if you think it worth then I can adapt...

Reply:
I think it's fine. Let's keep it as it is.